spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CAddTable.scala

```diff
@@ -38,6 +41,31 @@ class CAddTable[T: ClassTag, D: ClassTag](val inplace: Boolean = false)(
 
   output = Tensor[D]()
 
+  private def canFastBroadcast(tensor: Tensor[D], other: Tensor[D]): Boolean = {
+    if (tensor.nDimension != other.nDimension()) {
+      return false
+    }
+    var d = other.nDimension()
+    while(d > 0) {
+      if (tensor.size(d) != 1 && tensor.size(d) != other.size(d)) {
+        return false
+      }
+      d -= 1
+    }
+    return true
+  }
+
+  private def sumAlongDims(tensor: Tensor[D], other: Tensor[D]): Tensor[D] = {
+    val size = tensor.size()
+    var target: Tensor[D] = other
```

Comment:
why not just use `other` ?

Reply:
result of tensor sum is another tensor, not itself. Just now I found tensor sum will create new tensors, so add some code refactor.